geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementResult.java

```diff
@@ -113,6 +114,10 @@ public StatusCode getStatusCode() {
     return result;
   }
 
+  public <R extends RuntimeCacheElement> List<R> getResult(Class<R> clazz) {
+    return result.stream().map(clazz::cast).collect(Collectors.toList());
+  }
+
```

Comment:
until we have a constraint on the relationship between T and R, this doesn't actually provide any type safety...

Reply:
True, this only gives us the convenience for getResult to return us something we wanted instead of just RuntimeCacheElement. But that is what the other solutions can only give us as well.

Reply:
I searched in vain to be able to do something like:
```
<R extends T & RuntimeCacheElement> List<R> list(T filterConfig, CacheConfig existing);
```
this would constrain us on the relationship between T and R.